account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,406 @@
+                "redirect URL:- {}, Path url:- {}".format(self.hmrc_configuration.redirect_url, self.path)
+            )
+        elif endpoint_record.name == "mtd_hello_application_endpoint":
+            self.which_button_type_clicked = "application"
+            self.path = "/hello/application"
+            version = self._json_command('version')
+            self._logger.info(
+                "Connection button Clicked - endpoint name {},".format(self.name) +
+                "redirect URL:- {}, Path url:- {}".format(self.hmrc_configuration.redirect_url, self.path)
+            )
+        elif endpoint_record.name == "mtd_hello_user_endpoint":
+            self.which_button_type_clicked = "user"
+            self.path = "/hello/user"
+            self._logger.info(
+                "Connection button Clicked - endpoint name {}, ".format(self.name) +
+                "redirect URL:- {}, Path url:- {}".format(self.hmrc_configuration.redirect_url, self.path)
+            )
+            # search for token record for the API
+            token_record = self.env['mtd.api_tokens'].search([('api_id', '=', self.api_name.id)])
+            self._logger.info(
+                "Connection button Clicked - endpoint name {} , ".format(self.name) +
+                "and the api is :- {} ".format(self.api_name)
+            )
+            if not token_record.access_token and not token_record.refresh_token:
+                self._logger.info(
+                    "Connection button Clicked - endpoint name {0}, No access token ".format(self.name) +
+                    "found and no refresh_token found from the token record table."
+                )
+                api_tracker = self.env['mtd.api_request_tracker'].search([
+                    ('request_sent', '=', True),
+                    ('response_received', '=', False)
+                ])
+                self._logger.info(
+                    "Connection button Clicked - endpoint name {}, ".format(self.name) +
+                    "Checking to see if a request is in process"
+                )
+                time_10_mins_ago = (datetime.now() - timedelta(minutes=10))
+                format_time_10_mins_ago = time_10_mins_ago.strftime('%Y-%m-%d- %H:%M:%S')
+                if format_time_10_mins_ago >= api_tracker.create_date:
+                    # this means that time is more than 10 minutes a go so we can place a new request
+                    # and close this request.
+                    api_tracker.response_received = True
+                    self._logger.info(
+                        "Connection button Clicked - endpoint name {}, ".format(self.name) +
+                        "no Pending requests"
+                    )
+                    return self.get_user_authorisation()
+                else:
+                    # THe request made was within 10 mins so the user has to wait.
+                    test = True
```

Comment:
What does the `test` variable do?

Reply:
This has already been removed in my previous commits.